qmule/src/org/qcmg/qmule/FixCutAdapter.java

```diff
@@ -40,15 +40,11 @@ public static void main(final String[] argv) {
     }
 	@Override
 	protected int doWork() {
-		@SuppressWarnings("resource")
-		BufferedReader reader = INPUT == null ? 
-				new BufferedReader(new InputStreamReader(System.in)) : IOUtil.openFileForBufferedReading(INPUT);
-				
-		@SuppressWarnings("resource")
-		FastqWriter writer =  OUTPUT == null ?
-				null : (new FastqWriterFactory()).newWriter(OUTPUT);
 		 			        
-        try{
+        try( BufferedReader reader = INPUT == null ? 
+				new BufferedReader(new InputStreamReader(System.in)) : IOUtil.openFileForBufferedReading(INPUT);
+        	FastqWriter writer =  OUTPUT == null ?
+				null : (new FastqWriterFactory()).newWriter(OUTPUT);) {
```

Comment:
For symmetry you could create the stdout writer here too...
```
writer = OUTPUT == null ? new BufferedWriter(new OutputStreamWriter(System.out)) : ...
```
and then below you don't have to do the test if OUTPUT == null then `println`

Reply:
or better:
```
FastqWriter writer = OUTPUT == null ? (new BasicFastqWriter(System.out)) : (new FastqWriterFactory()).newWriter(OUTPUT);)
```

Reply:
sorry, I misunderstand it. now I changed to FastqWriter with BasicFastqWriter